gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/internal/WarnAboutComposeWithSerialization.kt

```diff
@@ -0,0 +1,28 @@
+package org.jetbrains.compose.internal
+
+import org.gradle.api.Project
+import org.gradle.configurationcache.extensions.serviceOf
+import org.gradle.internal.logging.text.StyledTextOutput
+import org.gradle.internal.logging.text.StyledTextOutputFactory
+
+internal fun Project.checkAndWarnAboutComposeWithSerialization() {
+    afterEvaluate {
```

Comment:
A [recommended way](https://github.com/Kotlin/kotlinx.serialization#using-the-plugins-block) to set up serialization is to use the dedicated Gradle plugin, so instead of checking a plugin classpath, you could just add a callback on the serialization plugin use:
```
project.plugins.withId("org.jetbrains.kotlin.plugin.serialization") {
     // ...
}
```